### PR TITLE
Add org.cataclysmdda.CataclysmDDA

### PR DESCRIPTION
**Pull request by grisha765**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
  
  - A turn-based survival game set in a post-apocalyptic world

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:https://github.com/CleverRaven/Cataclysm-DDA/issues/33682#issuecomment-527567273**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
> Please describe the application briefly.

This is missing

File: org.Cataclysm.DDA.Tiles.desktop [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Name=Cataclysm: Dark Days Ahead - Tiles
+GenericName=Post-apocalyptic roguelike
+Comment=A turn-based survival game set in a post-apocalyptic world.
+Icon=org.Cataclysm.DDA.Tiles
+Type=Application
+Exec=/app/bin/launch-cataclysm-dda
```
Comment (hfiguiere) [inline org.Cataclysm.DDA.Tiles.desktop:7]:
```suggestion
Exec=launch-cataclysm-dda
```

never a full path.

File: org.Cataclysm.DDA.Tiles.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+id: "org.Cataclysm.DDA.Tiles"
+runtime: "org.freedesktop.Platform"
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: /app/bin/launch-cataclysm-dda
+finish-args: 
+finish-args: 
+  [
```
Comment (hfiguiere) [inline org.Cataclysm.DDA.Tiles.yml:8]:
```suggestion
finish-args: 
  [
```

File: org.Cataclysm.DDA.Tiles.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+id: "org.Cataclysm.DDA.Tiles"
+runtime: "org.freedesktop.Platform"
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: /app/bin/launch-cataclysm-dda
```
Comment (hfiguiere) [inline org.Cataclysm.DDA.Tiles.yml:5]:
```suggestion
command: launch-cataclysm-dda
```

File: org.Cataclysm.DDA.Tiles.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+finish-args: 
+finish-args: 
+  [
+    "--filesystem=xdg-config/cataclysm-dda:create",
```
Comment (hfiguiere) [inline org.Cataclysm.DDA.Tiles.yml:9]: it shouldn't need this permission

File: org.Cataclysm.DDA.Tiles.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+modules:
+  - name: cataclysm-dda
+    buildsystem: simple
+    build-commands:
+      - tar -xvf cdda-linux-tiles-x64-2023-03-01-0054.tar.gz -C /app
+      - rm cdda-linux-tiles-x64-2023-03-01-0054.tar.gz
+      - mkdir -p /app/bin
+      - |
+        cat > /app/bin/launch-cataclysm-dda << EOF
```
Comment (hfiguiere) [inline org.Cataclysm.DDA.Tiles.yml:25]: make this either a separate file. or a `script` source. Not inline in the build commands.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
there doesn't seem to be an appstream file installed.

File: org.Cataclysm.DDA.Tiles.yml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        [ -d "\$XDG_CONFIG_HOME/cataclysm-dda/sound" ] || mkdir -p \$XDG_CONFIG_HOME/cataclysm-dda/sound
+        cp -r /app/cataclysmdda-0.F/data/sound/* \$XDG_CONFIG_HOME/cataclysm-dda/sound/
+
+        cd /app/cataclysmdda-0.F && ./cataclysm-launcher --configdir \$XDG_CONFIG_HOME/cataclysm-dda/config --userdir \$XDG_CONFIG_HOME/cataclysm-dda
+        EOF
+      - chmod +x /app/bin/launch-cataclysm-dda
+      - install -Dm644 org.Cataclysm.DDA.Tiles.svg /app/share/icons/hicolor/128x128/apps/org.Cataclysm.DDA.Tiles.svg
+      - install -Dm755 org.Cataclysm.DDA.Tiles.desktop /app/share/applications/org.Cataclysm.DDA.Tiles.desktop
+    sources:
+      - type: file
```
Comment (hfiguiere) [inline org.Cataclysm.DDA.Tiles.yml:47]: this should be `type: archive`. and then it will be extracted automatically.

File: org.Cataclysm.DDA.Tiles.yml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    "--share=ipc",
+    "--device=dri"
+  ]
+
+modules:
+  - name: cataclysm-dda
+    buildsystem: simple
+    build-commands:
+      - tar -xvf cdda-linux-tiles-x64-2023-03-01-0054.tar.gz -C /app
+      - rm cdda-linux-tiles-x64-2023-03-01-0054.tar.gz
```
Comment (hfiguiere) [inline org.Cataclysm.DDA.Tiles.yml:22]:
```suggestion
```

File: flathub.json [hunk 8]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: this is disappointing. why not build from source?

File: org.Cataclysm.DDA.Tiles.yml [hunk 9]
```diff
@@ -0,0 +1,54 @@
+id: "org.Cataclysm.DDA.Tiles"
```
Comment (hfiguiere) [inline org.Cataclysm.DDA.Tiles.yml:1]: `Cataclysm.org` doesn't resolve.

But you said

> The domain used for the application ID is [controlled by the application developers](https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository) and the [application id guidelines](https://docs.flathub.org/docs/for-app-authors/requirements#application-id) are followed.

Comment (hfiguiere) [reply org.Cataclysm.DDA.Tiles.yml:1]: maybe you meant `org.cataclysmdda` ?

image

Comment (grisha765) [reply org.Cataclysm.DDA.Tiles.yml:1]: Yes, I accidentally made a mistake

File: org.Cataclysm.DDA.Tiles.yml (hunk 4 above)
Comment (hfiguiere) [reply org.Cataclysm.DDA.Tiles.yml]: this wasn't addressed

Comment (grisha765) [reply org.Cataclysm.DDA.Tiles.yml]: This permission is required for the program to be able to create and view configuration folders and files in the user directory ~/.config/cataclysm-dda

Comment (hfiguiere) [reply org.Cataclysm.DDA.Tiles.yml]: Are you sure? reading the launch script, it doesn't. And if it really needs it, it should be using `--persist` instead.

File: org.cataclysmdda.tiles.yml [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+command: launch-cataclysm-dda
+finish-args: 
+  [
+    "--filesystem=xdg-config/cataclysm-dda:create",
+    "--socket=pulseaudio",
+    "--socket=wayland",
+    "--socket=fallback-x11",
+    "--share=ipc",
+    "--device=dri"
+  ]
```
Comment (bbhtt) [inline org.cataclysmdda.tiles.yml:14]: why is this mixture of json and yaml?

**Review by bbhtt — CHANGES_REQUESTED** [review]
Icon, desktop file, screenshots and metainfo file should be submitted upstream https://github.com/CleverRaven/Cataclysm-DDA/

File: org.Cataclysm.DDA.Tiles.yml (hunk 4 above)
Comment (grisha765) [reply org.Cataclysm.DDA.Tiles.yml]: I followed your advice, and the directory is not created, i still need to use `--filesystem`.

Comment (hfiguiere) [reply org.Cataclysm.DDA.Tiles.yml]: What directory are you looking at. Given the value of `XDG_CONFIG_HOME` if you are looking at `.config` then you are looking in the wrong place.

**Comment by grisha765** [comment]
> Icon, desktop file, screenshots and metainfo file should be submitted upstream https://github.com/CleverRaven/Cataclysm-DDA/

https://github.com/CleverRaven/Cataclysm-DDA/pull/75524

File: org.cataclysmdda.CataclysmDDA.yml [hunk 11]
```diff
@@ -0,0 +1,41 @@
+id: "org.cataclysmdda.CataclysmDDA"
+runtime: "org.freedesktop.Platform"
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: launch-cataclysm-dda
+finish-args: 
+  - --filesystem=xdg-config/cataclysm-dda:create
```
Comment (bbhtt) [inline org.cataclysmdda.CataclysmDDA.yml:7]: Why is this needed?

File: org.cataclysmdda.CataclysmDDA.yml [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+modules:
+  - name: cataclysm-dda
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/bin
+      - mkdir -p /app/cataclysm-dda
+      - cp -r * /app/cataclysm-dda
+      - install -Dm755 launch-cataclysm-dda /app/bin/launch-cataclysm-dda
+      - install -Dm644 org.cataclysmdda.CataclysmDDA.svg /app/share/icons/hicolor/128x128/apps/org.cataclysmdda.CataclysmDDA.svg
+      - sed -i 's/^Exec=cataclysm-tiles/Exec=launch-cataclysm-dda/' org.cataclysmdda.CataclysmDDA.desktop
```
Comment (bbhtt) [inline org.cataclysmdda.CataclysmDDA.yml:23]: Use `desktop-file-edit` here

File: org.cataclysmdda.CataclysmDDA.yml [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+
+modules:
+  - name: cataclysm-dda
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/bin
+      - mkdir -p /app/cataclysm-dda
+      - cp -r * /app/cataclysm-dda
+      - install -Dm755 launch-cataclysm-dda /app/bin/launch-cataclysm-dda
+      - install -Dm644 org.cataclysmdda.CataclysmDDA.svg /app/share/icons/hicolor/128x128/apps/org.cataclysmdda.CataclysmDDA.svg
```
Comment (bbhtt) [inline org.cataclysmdda.CataclysmDDA.yml:22]:
```suggestion
      - install -Dm644 org.cataclysmdda.CataclysmDDA.svg /app/share/icons/hicolor/scalable/apps/org.cataclysmdda.CataclysmDDA.svg
```

Why is svg icons going in 128x128 ?

File: flathub.json (hunk 8 above)
Comment (bbhtt) [reply flathub.json:2]: this is unanswered.

File: org.cataclysmdda.CataclysmDDA.yml (hunk 12 above)
Comment (grisha765) [reply org.cataclysmdda.CataclysmDDA.yml]: > Use `desktop-file-edit` here

done

File: org.cataclysmdda.CataclysmDDA.yml [hunk 14]
```diff
@@ -0,0 +1,41 @@
+id: "org.cataclysmdda.CataclysmDDA"
+runtime: "org.freedesktop.Platform"
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: launch-cataclysm-dda
+finish-args: 
+  - --persist=cataclysm-dda
```
Comment (hfiguiere) [inline org.cataclysmdda.CataclysmDDA.yml:7]: again, why do you need the persist given all the back and forth in the review you said it writes in the correct XDG location.

Comment (grisha765) [reply org.cataclysmdda.CataclysmDDA.yml:7]: Without this argument, it is impossible to create the necessary configuration directories for the game. In the [launch-cataclysm-dda](https://github.com/flathub/flathub/pull/5488/files/5070099a24ff0d51efcd2736fc6c1650e04072e8#diff-1ca570f7b28f6aa19fde4885f5c02890a6e2c7e7d4e95df5a9654ad6ad5e3627) script, I explicitly set the game to read and save its data in the `$HOME/cataclysm-dda` directory. Without the `--persist=cataclysm-dda` flag, this directory was not being created, and the game couldn't save its data there. By default, if the save directory isn't explicitly set, the game tries to save in its own directory where the binary is located, which is `/app/cataclysm-dda`. However, this directory is read-only and cannot be used for saving data.

Comment (hfiguiere) [reply org.cataclysmdda.CataclysmDDA.yml:7]: Maybe you fix  the launch script. https://github.com/flathub/flathub/pull/5488/files#diff-1ca570f7b28f6aa19fde4885f5c02890a6e2c7e7d4e95df5a9654ad6ad5e3627

It previously used XDG_HOME but you changed it in https://github.com/flathub/flathub/pull/5488/commits/ddfbbe2ca2383237d6ce841aafc6881a269adba0

File: flathub.json (hunk 8 above)
Comment (grisha765) [reply flathub.json:2]: The game takes a long time to build from the source code, there are official builds, I don’t see the point in this.

Comment (grisha765) [reply flathub.json:2]: If the community sees a real need for ARM support, we may consider it in the future, but for now I think it makes more sense to use the official x86_64 builds.

Comment (hfiguiere) [reply flathub.json:2]: > The game takes a long time to build from the source code,  

this is irrelevant.

> I don’t see the point in this.

aarch64 support.

Comment (grisha765) [reply flathub.json:2]: Compiling from source can be time consuming, especially for a large project like this. Given that official builds already exist, it makes more sense to use them. Also, I don't have the hardware to test aarch64 builds, so focusing on official x86_64 builds seems like the most rational approach. If ARM support really becomes important to the community, we may consider it in the future.

**Comment by grisha765** [comment]
bot, build org.cataclysmdda.CataclysmDDA

**Comment by grisha765** [comment]
bot, build org.cataclysmdda.CataclysmDDA

File: org.cataclysmdda.CataclysmDDA.yml [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+      - install -Dm755 data/xdg/org.cataclysmdda.CataclysmDDA.desktop /app/share/applications/org.cataclysmdda.CataclysmDDA.desktop
+      - install -Dm644 data/xdg/org.cataclysmdda.CataclysmDDA.appdata.xml /app/share/metainfo/org.cataclysmdda.CataclysmDDA.appdata.xml
+      - sed -i "/#WARNINGS = -w/c WARNINGS = -w" Makefile
+      - make -j $FLATPAK_BUILDER_N_JOBS $MAKE_ARGS
+      - make $MAKE_ARGS localization
+      - make $MAKE_ARGS install
+    sources:
+      - type: git
+        url: https://github.com/CleverRaven/Cataclysm-DDA.git
+        branch: "0.G-branch"
```
Comment (hfiguiere) [inline org.cataclysmdda.CataclysmDDA.yml:30]: NEVER a branch. a tag and/or commit is required.

File: org.cataclysmdda.CataclysmDDA.yml [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
```
Comment (hfiguiere) [inline org.cataclysmdda.CataclysmDDA.yml:30]:
```suggestion
```

**Comment by grisha765** [comment]
bot, build org.cataclysmdda.CataclysmDDA

**Comment by grisha765** [comment]
bot, build org.cataclysmdda.CataclysmDDA

**Comment by grisha765** [comment]
bot, build org.cataclysmdda.CataclysmDDA

**Comment by grisha765** [comment]
Unfortunately, I don’t have the opportunity to check the aarch64 assembly, but it seems to be assembled.

**Comment by RustoMCSpit** [comment]
image

unverified?

**Comment by grisha765** [comment]
> image
> 
> unverified?

Yes, I have no idea why it is like this, the developers confirmed it, and even added a link to flathub in the readme, this is usually how they write it if the developer has not confirmed their program on flathub
